utils/mail.go

```diff
@@ -228,14 +225,22 @@ func (e *Email) Send() error {
 	to := make([]string, 0, len(e.To)+len(e.Cc)+len(e.Bcc))
 	to = append(append(append(to, e.To...), e.Cc...), e.Bcc...)
 	// Check to make sure there is at least one recipient and one "From" address
-	if e.From == "" || len(to) == 0 {
-		return errors.New("Must specify at least one From address and one To address")
+	if len(to) == 0 {
+		return errors.New("Must specify at least one To address")
 	}
-	from, err := mail.ParseAddress(e.From)
+
+	from, err := mail.ParseAddress(e.Username)
 	if err != nil {
 		return err
 	}
-	e.From = from.String()
+
+	if len(e.From) == 0 {
+		e.From = e.Username
+	}
+	// use mail's RFC 2047 to encode any string
+	sub := mail.Address{Name: e.Subject, Address: ""}
+	e.Subject = strings.TrimRight(sub.String(), " <@>")
```

Comment:
I think this solution is trick. I still like the `mime.QEncoding`. could you do some research on go1.3, is there similar function.